src/tree/StaticWebAppTreeItem.ts

```diff
@@ -3,58 +3,44 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
+import { WebSiteManagementClient, WebSiteManagementModels } from "@azure/arm-appservice";
 import { ProgressLocation, window } from "vscode";
-import { AzExtTreeItem, AzureParentTreeItem, IActionContext, TreeItemIconPath } from "vscode-azureextensionui";
+import { AzExtTreeItem, AzureParentTreeItem, createAzureClient, IActionContext, TreeItemIconPath } from "vscode-azureextensionui";
 import { productionEnvironmentName } from '../constants';
 import { ext } from "../extensionVariables";
+import { getResourceGroupFromId, pollAzureAsyncOperation } from "../utils/azureUtils";
 import { getRepoFullname } from '../utils/gitHubUtils';
 import { localize } from "../utils/localize";
+import { nonNullProp } from "../utils/nonNull";
 import { openUrl } from '../utils/openUrl';
-import { requestUtils } from "../utils/requestUtils";
 import { treeUtils } from "../utils/treeUtils";
-import { EnvironmentTreeItem, StaticEnvironment } from './EnvironmentTreeItem';
+import { EnvironmentTreeItem } from './EnvironmentTreeItem';
 import { IAzureResourceTreeItem } from './IAzureResourceTreeItem';
 
-// using a custom defined type because the type provided by WebsiteManagementModels.StaticSiteARMResource doesn't match the actual payload
-export type StaticWebApp = {
-    id: string;
-    location: string;
-    name: string;
-    properties: {
-        defaultHostname: string;
-        repositoryUrl: string;
-        branch: string;
-        customDomains: string[];
-    };
-    sku: {
-        name: string;
-        tier: string;
-    };
-    // tslint:disable-next-line:no-reserved-keywords
-    type: string;
-};
-
 export class StaticWebAppTreeItem extends AzureParentTreeItem implements IAzureResourceTreeItem {
     public static contextValue: string = 'azureStaticWebApp';
     public readonly contextValue: string = StaticWebAppTreeItem.contextValue;
-    public readonly data: StaticWebApp;
+    public readonly data: WebSiteManagementModels.StaticSiteARMResource;
     public readonly childTypeLabel: string = localize('environment', 'Environment');
 
-    constructor(parent: AzureParentTreeItem, ss: StaticWebApp) {
+    public name: string;
+    public id: string;
+    public resourceGroup: string;
+    public label: string;
+    public repositoryUrl: string;
+    public branch: string;
+    public defaultHostname: string;
+
+    constructor(parent: AzureParentTreeItem, ss: WebSiteManagementModels.StaticSiteARMResource) {
         super(parent);
         this.data = ss;
```

Comment:
Data is a generic object that needs to be defined for this interface: https://github.com/microsoft/vscode-azurestaticwebapps/blob/main/src/tree/IAzureResourceTreeItem.ts

It's used by `viewProperties` https://github.com/microsoft/vscode-azurestaticwebapps/blob/main/src/commands/viewProperties.ts to display the Azure JSON object within VS Code.  This can be useful for looking at properties of Azure resources without having to list everything on the tree or visiting the portal.

I started using this command once I saw Eric implement it in Resource Groups.